tutorials/serverless-grafana-with-iap/index.md

```diff
@@ -0,0 +1,150 @@
+  * Explore and test the capabilities of Identity Aware Proxy (IAP) to restrict access to your Grafana dashboard.
+
+## Costs
+
+We'll be using billable components of Google Cloud, including the following:
+  * [Cloud Run](https://cloud.google.com/run)
+  * [CloudSQL](https://cloud.google.com/sql)
+  * [Cloud Load Balancer](https://cloud.google.com/load-balancing)
+  * [Identity-Aware Proxy](https://cloud.google.com/iap)
+  * [Secret Manager](https://cloud.google.com/secret-manager)
+
+Use the [pricing calculator](https://cloud.google.com/products/calculator) to generate a cost estimate based on your projected usage.
+
+## What happens behind the scenes
+
+You can easily deploy the setup with minimal effort; most of it is done through a Terraform script. Here is what happens in the background:
+
+First, all required APIs are enabled, e.g., IAM, Cloud Run, Compute, IAP, SQL by the Terraform script. Hence you don't need to worry about this step.
+
+Grafana requires a database for storing data. Therefore, a CloudSQL instance is created. The password for the database user is placed in Secret Manager, for secured access.
+
+Then, the Cloud Run container is deployed using the GCR mirror of the Grafana container image and started. The script also passes required environment variables to the container, such as information about the database connection, and auth proxy. 
+
+To make sure your Grafana dashboard is useful, a datasource for Google Cloud Monitoring is mounted. We use [Grafanas provisioning mechanism](https://grafana.com/docs/grafana/latest/administration/provisioning/) for that, which discovers data sources and dashboards from the disk. Since Cloud Run instances don't have persistent volumes at the moment, we can't just place the datasource and dashboard file into the filesystem. As a workaround, the file is added as a secret to Secret Manager and then copied to the required location so Grafana can pick up the data source correctly. Et voila, you now have mock data in your dashboard! An alternative could be to use [GCS Fuse](https://cloud.google.com/run/docs/tutorials/network-filesystems-fuse), which allows mounting a GCS bucket into the filesystem, but this requires modifying the Docker image.
+
+To access your Grafana dashboard, Cloud Load Balancer is configured to service HTTPS traffic from CloudRun using a Serverless Network Endpoint Group (NEG) as backend service. Identity-Aware Proxy (IAP) is integrated with the Load Balancer backend service. Client ID and secret are passed to the Load Balancer configuration. IAP provides headers containing the authorization information to applications secured with it ([link to documentation](https://cloud.google.com/iap/docs/signed-headers-howto)). Grafana provides the [functionality](https://grafana.com/docs/grafana/latest/auth/auth-proxy/) to receive exactly such header information for authentication. 
+
+With other words: IAP and Grafana? A match made in heaven.
```

Comment:
Best to keep the text simple. I would remove this sentence.

Reply:
removed sentence